src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,202 @@
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createQuery = "INSERT INTO cars (model, registration_number, manufacturer_id) "
+                + "VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement
+                        = connection.prepareStatement(createQuery,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, car.getModel());
+            createStatement.setString(2, car.getRegistrationNumber());
+            createStatement.setLong(3, car.getManufacturer().getId());
+            createStatement.executeUpdate();
+            ResultSet resultSet = createStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+                addDriversToCar(car);
+            }
+            return car;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't create car: " + car, throwable);
+        }
+    }
+
+    @Override
+    public Car get(Long id) {
+        String getQuery = "SELECT c.id AS car_id, model, registration_number, "
+                + "m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(getQuery)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarByManufacturer(resultSet);
+            }
+            return car;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get car by id = " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllQuery = "SELECT c.id AS car_id, model, registration_number, "
+                + "m.id AS manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAllQuery)) {
+            List<Car> cars = new ArrayList<>();
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarByManufacturer(resultSet));
+            }
+            for (Car car : cars) {
+                car.setDrivers(getAllDriversByCarId(car.getId()));
+            }
+            return cars;
```

Comment:
do this after catch block not to keep two connections open

Reply:
You can use cars.forEach()

Reply:
used forEach ()